## 13_itunes10.dpatch by  <maarten@informaarten.nl>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add aeMK to items-query to fix incompatibility issue with iTunes 10

--- src/plugins/out-daap-proto.c   2007-08-10 06:07:23.000000000 +0200
+++ src/plugins/out-daap-proto.c    2010-09-08 01:01:22.000000000 +0200
@@ -582,6 +582,7 @@
          */
 
         size = 8; /* mlit */
+        size += 9; /* aeMK for iTunes 10 */
         if(daap_wantsmeta(pinfo->meta, metaItemKind))
             /* mikd */
             size += 9;
@@ -777,6 +778,7 @@
         if(daap_wantsmeta(pinfo->meta, metaSongDataKind))
             current += dmap_add_char(current,"asdk",
                                         (char)atoi(valarray[SG_DATA_KIND]));
+        current += dmap_add_char(current,"aeMK",(char)0x1); // FIX for iTunes 10; 1 = Music
         if(EMIT(valarray[13]) && daap_wantsmeta(pinfo->meta, metaSongDataURL))
             current += dmap_add_string(current,"asul",valarray[SG_URL]);
         if(EMIT(valarray[5]) && daap_wantsmeta(pinfo->meta, metaSongAlbum))
